### DISCUSSION TITLE
Unexpected data type: <class 'backtrader.feeds.pandafeed.PandasData'>

### DISCUSSION THREAD

**Discussion by irusya930** [verybadsoldier/backtrader_plotting#77] [General]
Hello! Got an error trying pass PandasFeed data format. Running standard bt code:
```


    data = bt.feeds.PandasData(dataname = ethusdt_1m)

    # Add a strategy
    cerebro.addstrategy(TestStrategy)

    # Datas are in a subfolder of the samples. Need to find where the script is
    # because it could have been called from anywhere

    # Add the Data Feed to Cerebro
    cerebro.adddata(data)

    # Set our desired cash start
    cerebro.broker.setcash(1000000)

    # Add a FixedSize sizer according to the stake
    cerebro.addsizer(bt.sizers.FixedSize, stake=10)

    # Set the commission
    cerebro.broker.setcommission(commission=0.0)

    # Print out the starting conditions
    print('Starting Portfolio Value: %.2f' % cerebro.broker.getvalue())

    # Run over everything
    cerebro.run()

    # Print out the final result
    print('Final Portfolio Value: %.2f' % cerebro.broker.getvalue())


```

and then trying run:
'b = backtrader_plotting.Bokeh(style = 'bar')
cerebro.plot(b)'

which raises RuntimeError: Unexpected data type: <class 'backtrader.feeds.pandafeed.PandasData'>

can i somehow avoid it?

**Comment by verybadsoldier** [comment]
You can probably avoid it by assigning a custom name when adding the data to cerebro.

**Reply by vyouzhis** [reply]
as this:  cerebro.adddata(data, name="ok")
it's run ok

**Reply by PShiw** [reply]
Worked after adding name to data...

**Comment by thekie** [comment]
Hey. 
The problem might be in labelizer.py:44
```
val = getattr(data, n, "")
```

`val` can never be `None` because `default` in `getattr` is set to `""`. But the lines afterwards need it to be `None` in order to identify missing label names, if I understand this code correctly.

```
if val is None:
    continue
```

Cheers
